stingray/covariancespectrum.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import division
+
+import numpy as np
+
+from stingray import Lightcurve
+import stingray.utils as utils
+
+
+class Covariancespectrum(object):
+
+    def __init__(self, event_list, bin_width, band_interest=None,
+                 ref_band_interest=None):
+        """
+        Parameters
+        ----------
+        event_list : numpy 2D array
+            A numpy 2D array with first column as time of arrival and second
+            column as photon energies associated.
+
+        bin_width : float
+            The width of the energy bin when the spectrum is divided into N
+            divisions.
+
+        band_interest : iterable of tuples, default All
+            An iterable of tuples with minimum and maximum values of the range
+            in the band of interest.
+
+        ref_band_interest : iterable of tuples, default All
+            An iterable of tuples with minimum and maximum values of the range
+            in the band of interest in reference channel.
+        """
+        min_energy, max_energy = min(event_list.T[1]), max(event_list.T[1])
+        min_time, max_time = min(event_list.T[0]), max(event_list.T[0])
+
+        if band_interest is None:
+            band_interest = (min_energy, max_energy)
+
```

Comment:
Ah! This should work if we assume that none of the timestamp is missed in the event list, otherwise we may get not a single value from the _step_ array above in the `arange`.

Although, having used few if statements, I've dealt with the situation as two separate conditions. :)